### PR TITLE
Feat/7 import from sql schema

### LINKED ISSUE #7: Import from SQL schema
A good complement would be to be able to import from a **SQL schema / dump**, the same way one can export to SQL.
It could help bootstraping the creation of a schema, or just visualize an existing database.

### PR DESCRIPTION
**Pull request by DefinitelyNotAnAssassin**
Implementation of Feature Request #7 

1. Modified the Open FIle Button to be a Dropdown Button (Import Database, Import Diagram) 
2. Handled .db, .database, .sqlite, .sqlite3 transformation to ``TNode`` and ``TEdge`` using the ``ImportHelper.ts`` and ``NodeEdgeHelper.ts``

## ImportHelper.js 

--- 

```ts
export const importHelper = async (file: Uint8Array) => {
const sql = await initSqlJs();
const db = new sql.Database(file);
const database = db.exec('SELECT tbl_name FROM sqlite_master');
...
const dataTypeRegex = /^(.+?)(?:\((\d+)\))?$/;
...

return [nodes, edges]
}
```

This function is called on the ``useIPCListeners.ts`` to translate the File opened by the user to a proper TypeScript object. Returns a nodes and edges object that will be translated into TNode and TEdge. 

``useIPCListener.ts``

---

```ts
window.electron.ipcRenderer.on(
            'schemaOpened',
            async (_, file: Uint8Array) => {
                if (!vueFlow) return;
                historyStore.$reset();
                fileStore.$reset();
                const contents = await importHelper(file);
                const [nodes, edges] = contents;
                createNodeFromImport(nodes), createEdgeFromImport(edges, nodes);
            },
        );
```

This receives the file, pass it to the ``importHelper`` function, and utilizes the nodes and edges returned to call the appropriate functions that will translate the returned data to TNode and TEdge.

Caveats: 

1. Auto Layout is not working

### CODE REVIEW

File: src/renderer/src/components/Modules/Canvas/Canvas.vue [hunk 1]
```diff
@@ -22,7 +22,7 @@ import { VueFlow, useVueFlow } from '@vue-flow/core';
 import { nextTick, computed } from 'vue';
 
 const testElements = TestNodes;
-const testEdges = TestEdges;
+const testEdges = [];
```
Comment (Finestwork) [inline src/renderer/src/components/Modules/Canvas/Canvas.vue]: Please return the TestEdges, as we are currently in development mode.

File: src/renderer/src/composables/Miscellaneous/useIPCListeners.ts [hunk 2]
```diff
@@ -33,6 +37,18 @@ export function useIPCListeners() {
...
+        window.electron.ipcRenderer.on(
+            'schemaOpened',
+            async (_, file: Uint8Array) => {
+                if (!vueFlow) return;
+                historyStore.$reset();
+                fileStore.$reset();
+                const contents = await importHelper(file);
+                const [nodes, edges] = contents;
+                createNodeFromImport(nodes), createEdgeFromImport(edges, nodes);
+            },
```
Comment (Finestwork) [inline src/renderer/src/composables/Miscellaneous/useIPCListeners.ts:62]: You can call the `autoLayout()` function here, which is located in the 'useNodeAutoLayout' composable file. Please note that you should use async/await to achieve this. This ensures that Vue renders all of the nodes first without concerning how they appear on the canvas. Once the rendering process is complete, you can then execute `autoLayout()`.


Something like this:
```ts
async (_, file: Uint8Array) => {
    if (!vueFlow) return;
    historyStore.$reset();
    fileStore.$reset();
    const contents = await importHelper(file);
    const [nodes, edges] = contents;
    createNodeFromImport(nodes), createEdgeFromImport(edges, nodes);
    
    await nextTick();
    runAutoLayout();
},
```

**Review by Finestwork — COMMENTED** [review]
Goooood job, @DefinitelyNotAnAssassin ! Ready to merge it; just waiting for the changes. :fire:

File: src/renderer/src/composables/Miscellaneous/useIPCListeners.ts (hunk 2 above)
Comment (DefinitelyNotAnAssassin) [reply src/renderer/src/composables/Miscellaneous/useIPCListeners.ts:62]: autoLayout in the useIPCListeners file will cause a Pinia Error "... pinia is called before initialized"

Comment (Finestwork) [reply src/renderer/src/composables/Miscellaneous/useIPCListeners.ts:62]: May I see a code snippet of how you did it?

**Comment by DefinitelyNotAnAssassin** [comment]
Updated with the following functionalities: 

## Import From Database
--- 
- ``ImportHelper/importSchema``
- ``UseIPCListeners/on..'schemaOpened``


This utilizes SQL.js to query sqlite database file and transform the schema into the diagram. 

## Import From SQL / DDL
--- 

- ``ImportHelper/importDDL``
- ``UseIPCListeners/on..'ddlOpened``


This utilized the package ``sql-ddl-to-json-schema`` mentioned at #9.

**Pull request merged**